calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/camera_models/ceres_camera_model.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 TIER IV, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import List
+from typing import Optional
+
+from ceres_intrinsic_camera_calibrator.ceres_intrinsic_camera_calibrator_py import calibrate
+from intrinsic_camera_calibrator.camera_models.camera_model import CameraModel
+from intrinsic_camera_calibrator.camera_models.opencv_camera_model import OpenCVCameraModel
+import numpy as np
+
+
+class CeresCameraModel(CameraModel):
+    """Basic Ceres camera model class."""
+
+    def __init__(
+        self,
+        k: Optional[np.array] = None,
+        d: Optional[np.array] = None,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+    ):
+        super().__init__(k, d, height, width)
+        self.radial_distortion_coefficients: int = None
+        self.rational_distortion_coefficients: int = None
+        self.use_tangential_distortion: bool = None
+        self.pre_calibration_num_samples: int = None
+        self.regularization_weight: float = None
+        self.verbose = (
+            True if os.getenv("GLOG_minloglevel") == "0" else False
+        )  # cSpell:ignore minloglevel
+
+    def init_calibrate(
+        self, object_points_list: List[np.array], image_points_list: List[np.array]
+    ) -> OpenCVCameraModel:
+        """Init calibrate of Ceres camera model."""
+        camera_model = OpenCVCameraModel()
+        camera_model.update_config(
+            radial_distortion_coefficients=self.radial_distortion_coefficients,
+            rational_distortion_coefficients=self.rational_distortion_coefficients,
+            use_tangential_distortion=self.use_tangential_distortion,
+        )
+
+        # Consider only part of data (distributed uniformly)
```

Comment:
Is this assumption valid? (I do not remember implementing shuffling :thinking: )

Reply:
For initial calibration I assume it is better to take uniform distributed (from collecting sequence point of view) samples, e.g. If we have 80 samples and 40 samples limit for initial calibration, we take 1, 3, 5... instead of 1-40. We could take samples base on pose distribution as well, but I didn't find it necessary.
In general, I would prefer to remove this limit in favor of timeout parameter, but OpenCV API does not consider this. Separate thread for init calibration task would not work - breaking it will not result with recent intrinsic parameters. Maybe calibrating frame by frame here is an answer, but I doubt the performance.